roles/manage_agents/tasks/remove_agents.yml

```diff
@@ -0,0 +1,34 @@
+---
+- name: Number of agents being removed
+  ansible.builtin.set_fact:
+    managed_agents_removed_count: "{{ manage_agents_allocated.resources | length - manage_agents_desired_count | int }}"
+
+- name: Wait for all expected agents to be removed from the cluster
+  ansible.builtin.include_tasks:
+    file: allocated_agents.yml
+  with_sequence: start=1 end=10
+  loop_control:
+    pause: 30
+    break_when:
```

Comment:
the pattern `until`/`retries`/`delay` doesn't work with `include_tasks`, but loop pattern look to be supported.